File: rb/features.rb
# AirQuality SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module AirQualityFeatures
  def self.make_feature(name)
    case name
    when "base"
      AirQualityBaseFeature.new
    when "test"
      AirQualityTestFeature.new
    else
      AirQualityBaseFeature.new
    end
  end
end

File: py/features.py
# AirQuality SDK feature factory

from feature.base_feature import AirQualityBaseFeature
from feature.test_feature import AirQualityTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AirQualityBaseFeature(),
        "test": lambda: AirQualityTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
